### DISCUSSION TITLE
Nvidia - Cuda version too recent for PyTorch - Launch failure

### DISCUSSION THREAD

**Discussion by McAtty** [oobabooga/text-generation-webui#2013] [Q&A]
Hi there, I was wondering how you get the older versions of Cuda running. 

I found the Windows installer for the Nvidia Cuda system, but I'm not sure which environment the Cuda install has to be targeting (How to get it for the Git Bash install instead of the Windows install. Does the system targeted by the install not matter because it's both the same functional target?)

Essentially the error I get says that the Cuda version does not match the version inside the PyTorch bake.

**Comment by Dhaladom** [comment]
Cuda usually comes with your nvidia driver. Try do downgrade it.

**Comment by BennettFourr** [comment]
This is the link for the cuda version needed for textgenwebui   [Nvidia](https://developer.nvidia.com/cuda-11-7-0-download-archive)

**Comment by WildBirdTG** [comment]
On my Win11 the CUDA come with the driver is 12.1, you can install 11.7 without uninstall 12.1, however on my laptop when both versions existed, Windows crashed when I run torch.cuda.is_availabe(), so highly recommend to uninstall 12.1 and install 11.7. Also it's better to make sure your torch is the compatible version:
`pip uninstall torch`
`pip cache purge`
`pip3 install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu117`

This is from this discussion and works on my Windows11 pro with 4060 laptop
https://github.com/openai/whisper/discussions/47
